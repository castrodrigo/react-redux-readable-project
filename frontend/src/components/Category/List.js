import React from "react";
import Category from "../../containers/Category";

const List = ({ title, categories }) => (
  <div>
    <h2>{title}</h2>
    <ul>
      {categories.map(path => (
        <li key={path}>
          <Category path={path} />
        </li>
      ))}
    </ul>
  </div>
);

export default List;
