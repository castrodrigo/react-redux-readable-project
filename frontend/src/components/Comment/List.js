import React from "react";
import Comment from "../../containers/Comment";

const List = ({ title, commentIds }) => (
  <div>
    <h2>{title}</h2>
    <ul>
      {commentIds &&
        commentIds.map(id => (
          <li key={id}>
            <Comment id={id} />
          </li>
        ))}
    </ul>
  </div>
);

export default List;
