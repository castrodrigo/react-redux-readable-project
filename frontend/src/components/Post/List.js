import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Post from "../../containers/Post/Post";

const Title = styled.h3`
  display: inline;
`;

const ListItem = styled.li`
  margin-bottom: 0.75em;
`;

const List = ({ title, postIds }) => (
  <div>
    <Title>{title}</Title>
    <ul>
      {postIds.map(id => (
        <ListItem key={id}>
          <Post id={id} />
        </ListItem>
      ))}
    </ul>
  </div>
);

List.propTypes = {
  title: PropTypes.string.isRequired,
  postIds: PropTypes.array
};

export default List;
