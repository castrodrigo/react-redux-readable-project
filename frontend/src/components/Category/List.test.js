import React from "react";
import { shallow } from "enzyme";
import List from "./List";
import categoryIds from "../__mockData__/categoryIds";

const title = "Test List";

describe("List Render", () => {
  it("Should render categories", () => {
    const wrapper = shallow(<List title={title} categories={categoryIds} />);

    expect(wrapper).toMatchSnapshot();
  });
});
