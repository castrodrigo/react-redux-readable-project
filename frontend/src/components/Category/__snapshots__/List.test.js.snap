// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Render Should render categories 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    categories={
      Array [
        "react",
        "redux",
        "udacity",
      ]
    }
    title="Test List"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Test List
        </h3>,
        <ul>
          <li>
            <Connect(ListItem)
              path="react"
            />
          </li>
          <li>
            <Connect(ListItem)
              path="redux"
            />
          </li>
          <li>
            <Connect(ListItem)
              path="udacity"
            />
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test List",
        },
        "ref": null,
        "rendered": "Test List",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Connect(ListItem)
                path="react"
              />
            </li>,
            <li>
              <Connect(ListItem)
                path="redux"
              />
            </li>,
            <li>
              <Connect(ListItem)
                path="udacity"
              />
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "react",
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ListItem)
                path="react"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "react",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "redux",
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ListItem)
                path="redux"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "redux",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "udacity",
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ListItem)
                path="udacity"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "udacity",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Test List
          </h3>,
          <ul>
            <li>
              <Connect(ListItem)
                path="react"
              />
            </li>
            <li>
              <Connect(ListItem)
                path="redux"
              />
            </li>
            <li>
              <Connect(ListItem)
                path="udacity"
              />
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test List",
          },
          "ref": null,
          "rendered": "Test List",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Connect(ListItem)
                  path="react"
                />
              </li>,
              <li>
                <Connect(ListItem)
                  path="redux"
                />
              </li>,
              <li>
                <Connect(ListItem)
                  path="udacity"
                />
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "react",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ListItem)
                  path="react"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "react",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "redux",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ListItem)
                  path="redux"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "redux",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "udacity",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ListItem)
                  path="udacity"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "udacity",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
